cdp_scrapers/instances/seattle.py

```
@@ -0,0 +1,359 @@
+# TODO: is it better to use uuids like EventItemMatterGuid for cdp *_ext_src_id?
+LEGISTAR_MATTER_EXT_ID      = 'EventItemMatterId'
+LEGISTAR_MATTER_TITLE       = 'EventItemTitle'
+LEGISTAR_MATTER_NAME        = 'EventItemMatterName'
+LEGISTAR_MATTER_TYPE        = 'EventItemMatterType'
+LEGISTAR_MATTER_STATUS      = 'EventItemMatterStatus'
+# Session.session_datetime is a combo of EventDate and EventTime
+LEGISTAR_SESSION_DATE       = 'EventDate'
+LEGISTAR_SESSION_TIME       = 'EventTime'
+LEGISTAR_BODY_NAME          = 'EventBodyName'
+LEGISTAR_FILE_EXT_ID        = 'MatterAttachmentId'
+LEGISTAR_FILE_NAME          = 'MatterAttachmentName'
+LEGISTAR_FILE_URI           = 'MatterAttachmentHyperlink'
+LEGISTAR_PERSON_EMAIL       = 'PersonEmail'
+LEGISTAR_PERSON_EXT_ID      = 'PersonId'
+LEGISTAR_PERSON_NAME        = 'PersonFullName'
+LEGISTAR_PERSON_PHONE       = 'PersonPhone'
+LEGISTAR_PERSON_WEBSITE     = 'PersonWWW'
+LEGISTAR_VOTE_DECISION      = 'VoteResult'
+LEGISTAR_VOTE_EXT_ID        = 'VoteId'
+
+LEGISTAR_EV_ITEMS           = 'EventItems'
+LEGISTAR_EV_SITE_URL        = 'EventInSiteURL'
+LEGISTAR_EV_ATTACHMENTS     = 'EventItemMatterAttachments'
+LEGISTAR_EV_VOTES           = 'EventItemVoteInfo'
+LEGISTAR_VOTE_PERSONS       = 'PersonInfo'
+
+CDP_VIDEO_URI   = 'video_uri'
+CDP_CAPTION_URI = 'caption_uri'
+
+
+# base class for scraper to convert legistar api data -> cdp ingestion model data
+# TODO: double-check unique class name in cdp to avoid confusion
+class LegistarScraper:
+    def __init__(self, client: str):
+        self.client_name = client
+
+
+    def is_legistar_compatible(self) -> bool:
+        '''
+        return True if can get successful legistar api response
+        '''
+        client = self.client_name
+
+        # simplest check, if the GET request works, it is a legistar place
+        try:
+            resp = urlopen(f'{LEGISTAR_BASE}/bodies')
+            return (resp.status == 200)
+        except:
```

Can't allow bare except statement.